component/azstorage/datalake.go

```diff
@@ -268,9 +226,8 @@ func (dl *Datalake) CreateFile(name string, mode os.FileMode) error {
 func (dl *Datalake) CreateDirectory(name string) error {
 	log.Trace("Datalake::CreateDirectory : name %s", name)
 
-	directoryURL := dl.Filesystem.NewDirectoryURL(filepath.Join(dl.Config.prefixPath, name))
-	_, err := directoryURL.Create(context.Background(), false)
-
+	directoryURL := dl.Filesystem.NewDirectoryClient(filepath.Join(dl.Config.prefixPath, name))
+	_, err := directoryURL.Create(context.Background(), &directory.CreateOptions{}) //TODO:: track2: false was a parameter after context for what?
```

Comment:
False was to prevent directory creation if it already exists. This adds the header `If-None-Match: "*"`.
Look for similar option in track2.